simple/directed.go

```diff
@@ -120,7 +120,6 @@ func (g *DirectedGraph) RemoveNode(n graph.Node) {
 	}
 	delete(g.predecessors, n.ID())
 
-	g.maxID-- // Fun facts: even if this ID doesn't exist this still works!
```

Comment:
Sorry, I'm wrong here. We need an ordered set of node. Fortunately a type exists for this, I'll add that, so your change here is fine.